grommet/grommet
# Data points normalization for Chart rendering

#### What does this PR do?
Provides normalization function in order to help Chart consumers to normalize a given series.
Currently, the chart component assumes the array of values (whether its for Line/Bar/Area/Axis) is always normalize and the points are evenly spreader on the array. This PR will address the problem of data given with blanks area or gaps by normalizing the x-axis and correspondingly the y-axis.

normalize method will normalize the values in case of data-gaps and blank areas, and will fill 'undefined' for the missing points added during normalization.

#### Where should the reviewer start?
Start with the tests of the documented examples, to first understand the required functionality of the normalization. and then drill down to the actual 'normalize' method the explore its behavior.

#### What testing has been done on this PR?
9 different tests to test normalization on different granularity and different arrays of series.

#### How should this be manually tested?
use unnormalized data on a Chart and see how the Chart is being rendered with and without the function.
   
#### Any background context you want to provide?
In many cases, in order to display analytics data, we are using data of inconsistent granularity between data points. This is what happening in our product, hence I created this function in order to provide a solution for the consumers of the Chart that uses data that might have gaps on the data.

#### What are the relevant issues?
Once this PR will be aproved, we'll need to address the Chart itself to handle 'undefined' for the missing data points.

#### Screenshots (if appropriate)

#### Do the grommet docs need to be updated?
yep, we'll need to expose the method as part of the charts functionality.

#### Should this PR be mentioned in the release notes?
Yep, its important to expose it.

#### Is this change backwards compatible or is it a breaking change?
Not breaking backwards compatibility, its all new functionality.
